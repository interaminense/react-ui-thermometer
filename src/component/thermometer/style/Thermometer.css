.thermometer .thermometer__mercury {
  transition: all .5s ease-in-out; }

.thermometer .thermometer__percent-current:before, .thermometer .thermometer__percent-current:after, .thermometer .thermometer__draw-b:after, .thermometer .thermometer__draw-b:before {
  content: "";
  margin: auto; }

.thermometer .thermometer__draw-a:after, .thermometer .thermometer__draw-a:before {
  content: "";
  background-color: #fff; }

.thermometer ul.thermometer__statistics {
  font-size: 11px;
  font-style: italic;
  font-weight: 500; }

.thermometer .thermometer__percent-current {
  font-size: 18px;
  font-style: normal;
  font-weight: 500; }

.thermometer ul.thermometer__statistics {
  margin: 0;
  padding: 0; }

.thermometer.thermometer--small .thermometer__draw-a {
  width: 14px; }
  .thermometer.thermometer--small .thermometer__draw-a:after {
    width: 34px;
    height: 34px;
    bottom: -12px; }

.thermometer.thermometer--small .thermometer__draw-b:before {
  width: 4px; }

.thermometer.thermometer--small .thermometer__draw-b:after {
  position: absolute;
  top: 13px;
  right: 0;
  bottom: initial;
  left: 0;
  width: 24px;
  height: 24px; }

.thermometer.thermometer--small .thermometer__meter {
  width: 4px; }

.thermometer.thermometer--normal .thermometer__draw-a {
  width: 30px; }
  .thermometer.thermometer--normal .thermometer__draw-a:after {
    width: 50px;
    height: 50px;
    bottom: -20px; }

.thermometer.thermometer--normal .thermometer__draw-b:before {
  width: 10px; }

.thermometer.thermometer--normal .thermometer__draw-b:after {
  position: absolute;
  top: 8px;
  right: 0;
  bottom: initial;
  left: 0;
  width: 34px;
  height: 34px; }

.thermometer.thermometer--normal .thermometer__meter {
  width: 10px; }

.thermometer.thermometer--large .thermometer__draw-a {
  width: 50px; }
  .thermometer.thermometer--large .thermometer__draw-a:after {
    width: 70px;
    height: 70px;
    bottom: -30px; }

.thermometer.thermometer--large .thermometer__draw-b:before {
  width: 30px; }

.thermometer.thermometer--large .thermometer__draw-b:after {
  position: absolute;
  top: 0;
  right: 0;
  bottom: initial;
  left: 0;
  width: 50px;
  height: 50px; }

.thermometer.thermometer--large .thermometer__meter {
  width: 30px; }

.thermometer {
  width: 100px;
  position: relative; }
  .thermometer ul.thermometer__statistics {
    position: absolute;
    top: 0;
    right: initial;
    bottom: initial;
    left: 0;
    height: 100%;
    text-shadow: 1px 1px #fff;
    z-index: 1; }
    .thermometer ul.thermometer__statistics li {
      position: absolute;
      top: initial;
      right: initial;
      bottom: initial;
      left: 0;
      width: 56px;
      text-align: right;
      list-style: none; }
      .thermometer ul.thermometer__statistics li:before {
        position: absolute;
        top: initial;
        right: initial;
        bottom: -1px;
        left: 0;
        content: "";
        width: 100%;
        background: linear-gradient(90deg, rgba(0, 0, 0, 0.2), transparent);
        height: 1px; }
  .thermometer .thermometer__meter {
    position: absolute;
    top: 12px;
    right: 0;
    bottom: initial;
    left: 0;
    height: calc(100% - 58px);
    margin: auto;
    background-color: #d6d6d6;
    border-radius: 50px 50px 0 0;
    z-index: 1; }
  .thermometer .thermometer__mercury {
    position: absolute;
    top: initial;
    right: initial;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 0;
    border-radius: 50px 50px 0 0; }
  .thermometer .thermometer__mask {
    position: absolute;
    top: initial;
    right: initial;
    bottom: 0;
    left: initial;
    overflow: hidden;
    width: 100%;
    height: 100%;
    border-radius: 50px 50px 0 0; }
  .thermometer .thermometer__bg-color {
    position: absolute;
    top: initial;
    right: initial;
    bottom: 0;
    left: initial;
    width: 100%;
    background: linear-gradient(#2196F3, #8BC34A, #f44336); }
  .thermometer .thermometer__percent-current {
    position: absolute;
    top: 4px;
    right: initial;
    bottom: initial;
    left: 0;
    transform: translateX(calc(-100% - 8px)) translateY(-50%);
    padding: 5px 10px;
    background-color: #fff;
    color: #000;
    border-radius: 2px;
    box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.1), 3px 3px 6px rgba(0, 0, 0, 0.1); }
    .thermometer .thermometer__percent-current:before, .thermometer .thermometer__percent-current:after {
      position: absolute;
      top: 0;
      right: 0;
      bottom: 0;
      left: initial;
      width: 0;
      height: 0; }
    .thermometer .thermometer__percent-current:after {
      border-left: 8px solid #fff;
      border-top: 6px solid transparent;
      border-bottom: 6px solid transparent;
      transform: translateX(calc(100% - 2px)) translateY(0); }
    .thermometer .thermometer__percent-current:before {
      border-left: 8px solid rgba(0, 0, 0, 0.1);
      border-top: 6px solid transparent;
      border-bottom: 6px solid transparent;
      transform: translateX(calc(100% - 0px)) translateY(0); }
  .thermometer .thermometer__draw-a {
    position: relative;
    height: calc(100% - 20px);
    margin: auto;
    background-color: #fff;
    border-radius: 50px 50px 0 0;
    box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.05), 1px 1px 5px rgba(0, 0, 0, 0.2); }
    .thermometer .thermometer__draw-a:before {
      position: absolute;
      top: initial;
      right: initial;
      bottom: 0;
      left: 0;
      height: 50px;
      width: 100%;
      z-index: 1; }
    .thermometer .thermometer__draw-a:after {
      position: absolute;
      top: initial;
      right: initial;
      bottom: -20px;
      left: -10px;
      width: 50px;
      border-radius: 50%;
      box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.05), 1px 1px 5px rgba(0, 0, 0, 0.2); }
  .thermometer .thermometer__draw-b {
    position: absolute;
    top: initial;
    right: 0;
    bottom: 0;
    left: 0;
    width: 50px;
    height: 50px;
    margin: auto;
    z-index: 1; }
    .thermometer .thermometer__draw-b:after {
      width: 24px;
      height: 24px;
      background-color: #f44336;
      border-radius: 50%; }
    .thermometer .thermometer__draw-b:before {
      position: absolute;
      top: 0;
      right: 0;
      bottom: 40px;
      left: 0;
      height: 20px;
      background-color: #f44336;
      border-radius: 50px 50px 0 0; }
