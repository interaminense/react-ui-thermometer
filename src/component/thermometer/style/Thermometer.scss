@import 'variables';
@import 'mixins';

%transition-a {
	transition: all .5s ease-in-out;
}

%extend-a {
	content: "";
	margin: auto;
}

%extend-b {
	content: "";
	background-color: $color-white;
}

%font-style-a {
	font-size: 11px;
	font-style: italic;
	font-weight: 500;
}

%font-style-b {
	font-size: 18px;
	font-style: normal;
	font-weight: 500;
}

%no-gap {
	margin: 0;
	padding: 0;
}

%thermometer-small {
	.thermometer__draw-a {
		width: $thermometer-small-w-drawA;
		
		&:after {
			width: $thermometer-small-drawA-after;
			height: $thermometer-small-drawA-after;
			bottom: -12px;
		}
	}
	.thermometer__draw-b {
		&:before {
			width: $thermometer-small-w-meter;
		}
		&:after {
			@include absolute-custom(13px, 0, initial, 0);
			width: $thermometer-small-drawB-after;
			height: $thermometer-small-drawB-after;
		}
	}
	.thermometer__meter {
		width: $thermometer-small-w-meter;
	}
}

%thermometer-normal {
	.thermometer__draw-a {
		width: $thermometer-normal-w-drawA;
		
		&:after {
			width: $thermometer-normal-drawA-after;
			height: $thermometer-normal-drawA-after;
			bottom: -20px;
		}
	}
	.thermometer__draw-b {
		&:before {
			width: $thermometer-normal-w-meter;
		}
		&:after {
			@include absolute-custom(8px, 0, initial, 0);
			width: $thermometer-normal-drawB-after;
			height: $thermometer-normal-drawB-after;
		}
	}
	.thermometer__meter {
		width: $thermometer-normal-w-meter;
	}
}

%thermometer-large {
	.thermometer__draw-a {
		width: $thermometer-large-w-drawA;
		
		&:after {
			width: $thermometer-large-drawA-after;
			height: $thermometer-large-drawA-after;
			bottom: -30px;
		}
	}
	.thermometer__draw-b {
		&:before {
			width: $thermometer-large-w-meter;
		}
		&:after {
			@include absolute-custom(0, 0, initial, 0);
			width: $thermometer-large-drawB-after;
			height: $thermometer-large-drawB-after;
		}
	}
	.thermometer__meter {
		width: $thermometer-large-w-meter;
	}
}

.thermometer {
	width: $thermometer-w;
	position: relative;

	&.thermometer--small {
		@extend %thermometer-small;
	}
	&.thermometer--normal {
		@extend %thermometer-normal;
	}
	&.thermometer--large {
		@extend %thermometer-large;
	}

	ul.thermometer__statistics {
		@include absolute-custom(0, initial, initial, 0);
		@extend %font-style-a;
		@extend %no-gap;
		height: 100%;
		text-shadow: 1px 1px $color-white;
		z-index: 1;

		li {
			@include absolute-custom(initial, initial, initial, 0);
			width: 56px;
			text-align: right;
			list-style: none;

			&:before {
				@include absolute-custom(initial, initial, -1px, 0);
				content: "";
				width: 100%;
				background: $thermometer-statistic-li-before;
				height: 1px;
			}
		}
	}

	.thermometer__meter {
		@include absolute-custom(12px, 0, initial, 0);
		height: calc(100% - 58px);
		margin: auto;
		background-color: $color-gray;
		border-radius: $border-radius-a;
		z-index: 1;
	}

	.thermometer__mercury {
		@include absolute-custom(initial, initial, 0, 0);
		@extend %transition-a;
		width: 100%;
		height: 0;
		border-radius: $border-radius-a;
	}

	.thermometer__mask {
		@include absolute-custom(initial, initial, 0, initial);
		overflow: hidden;
		width: 100%;
		height: 100%;
		border-radius: $border-radius-a;
	}

	.thermometer__bg-color {
		@include absolute-custom(initial, initial, 0, initial);
		width: 100%;
		background: $thermometer-bg-color;
	}

	.thermometer__percent-current {
		@include absolute-custom(4px, initial, initial, 0);
		@include translate(calc(-100% - 8px), -50%);
		@extend %font-style-b;
		padding: 5px 10px;
		background-color: $color-white;
		color: $color-black;
		border-radius: $border-radius-b;
		box-shadow: $box-shadow-b;

		&:before,
		&:after {
			@extend %extend-a;
			@include absolute-custom(0, 0, 0, initial);
			width: 0;
			height: 0;
		}

		&:after {
			@include border(left, 8px, $color-white);
			@include border(top, 6px, transparent);
			@include border(bottom, 6px, transparent);
			@include translate(calc(100% - 2px), 0);
		}

		&:before {
			@include border(left, 8px, rgba($color-black, .1));
			@include border(top, 6px, transparent);
			@include border(bottom, 6px, transparent);
			@include translate(calc(100% - 0px), 0);
		}
	}

	.thermometer__draw-a {
		position: relative;
		height: calc(100% - 20px);
		margin: auto;
		background-color: $color-white;
		border-radius: $border-radius-a;
		box-shadow: $box-shadow-a;

		&:after,
		&:before {
			@extend %extend-b;
		}
		&:before {
			@include absolute-custom(initial, initial, 0 ,0);
			height: 50px;
			width: 100%;
			z-index: 1;
		}
		&:after {
			@include absolute-custom(initial, initial, -20px ,-10px);
			width: 50px;
			border-radius: $box-shadow-c;
			box-shadow: $box-shadow-a;
		}
	}

	.thermometer__draw-b {
		@include absolute-custom(initial, 0, 0, 0);
		width: 50px;
		height: 50px;
		margin: auto;
		z-index: 1;

		&:after {
			@extend %extend-a;
			width: 24px;
			height: 24px;
			background-color: $color-a;
			border-radius: $box-shadow-c;
		}

		&:before {
			@extend %extend-a;
			@include absolute-custom(0, 0, 40px, 0);
			height: 20px;
			background-color: $color-a;
			border-radius: $border-radius-a;
		}
	}
}