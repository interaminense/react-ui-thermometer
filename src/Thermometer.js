import React from 'react';
import Proptypes from 'prop-types';
import './style.scss';

const getFormattedLabel = ({label, insertAfter}, val) => insertAfter ? val + label : label + val;

const getIntervals = (format, max, steps) => {
	let intervals = [];

	for (let step = 0; step <= steps; step++) {
		const val = Math.abs(((max / steps) * step).toFixed(2));
		const percent = (val / max) * 100;
		const interval = {
			percent,
			label: getFormattedLabel(format, val)
		};
		
		intervals.push(interval);
	}

	return intervals;
}

const Thermometer = ({format, height, max, size, steps, theme, value}) => {
	const intervals = getIntervals(format, max, steps);
	const percent = value / max * 100;
	const valstr = getFormattedLabel(format, value);

	const stepIntervals = intervals.map(({label, percent}, i) => (
		<li key={i} style={{ bottom: `calc(${percent}% - 1px)` }}>
			{label}
		</li>
	));

	const styledHeight = {height: `${height}px`};

	const styledHeightPercent = {height: `${percent}%`};

	const styledHeightBgColor = {height: `calc(${height}px - 57px)`};

	return (
		<div style={styledHeight} className={`thermometer thermometer--${size} thermometer--theme-${theme}`}>
			<div className="thermometer__draw-a"></div>
			<div className="thermometer__draw-b"></div>
			<div className="thermometer__meter">
				<ul className="thermometer__statistics">{stepIntervals}</ul>
				<div style={styledHeightPercent} className="thermometer__mercury">
					<div className="thermometer__percent-current">{valstr}</div>
					<div className="thermometer__mask">
						<div className="thermometer__bg-color" style={styledHeightBgColor}></div>
					</div>
				</div>
			</div>
		</div>
	);
};

Thermometer.propTypes = {
	format: Proptypes.shape({
		label: Proptypes.string,
		insertAfter: Proptypes.bool
	}),
	height: Proptypes.number,
	max: Proptypes.number,
	size: Proptypes.oneOf(['small', 'normal', 'large']),
	steps: Proptypes.number,
	theme: Proptypes.oneOf(['light', 'dark']),
	value: Proptypes.number
};

Thermometer.defaultProps = {
	format: {
		label: '%',
		insertAfter: true
	},
	height: 200,
	max: 100,
	size: 'small',
	steps: 4,
	theme: 'light',
	value: 0
};

export default Thermometer;