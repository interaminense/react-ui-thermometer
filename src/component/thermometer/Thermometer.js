import React, { Component } from 'react';
import './style/Thermometer.css';

class Thermometer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      max: this.props.max || 100, //default 100
      steps: this.props.steps || 4, //default 4
      format: this.props.format || '',
      size: () => {
        if (this.props.size !== 'small' && this.props.size !== 'normal' && this.props.size !== 'large') {
          return 'normal';
        }
        return this.props.size;
      },
      height: this.props.height || 200, //default 100
      value: this.props.value || 0, //default 0
      valstr: () => this.state.format + this.state.value,
      percent: () => this.state.value / this.state.max * 100,
      intervals: []
    }

    for (let step = 0; step <= this.state.steps; step++) {
      let val = ((this.state.max / this.state.steps) * step).toFixed(2);
      let percent = (val / this.state.max) * 100;
      let interval = { percent: percent, label: this.state.format + val };
      this.state.intervals.push(interval);
    }
  }

  render() {

    const stepIntervals = this.state.intervals.map((step, i) => {
      return (
        <li key={i} style={{ bottom: `calc(${step.percent}% - 1px)` }}>
          {step.label}
        </li>
      );
    });

    return (
      <div style={{ height: `${this.state.height}px` }} className={`thermometer thermometer--${this.state.size()}`}>
        <div className="thermometer__draw-a"></div>
        <div className="thermometer__draw-b"></div>
        <div className="thermometer__meter">
          <ul className="thermometer__statistics">{stepIntervals}</ul>
          <div style={{ height: `${this.state.percent()}%` }} className="thermometer__mercury">
            <div className="thermometer__percent-current">{this.state.valstr()}</div>
            <div className="thermometer__mask">
              <div className="thermometer__bg-color" style={{ height: `calc(${this.state.height}px - 57px)` }}></div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Thermometer;