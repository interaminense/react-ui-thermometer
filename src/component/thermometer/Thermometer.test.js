import React from 'react';
import Thermometer from './Thermometer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { shallow } from 'enzyme';

configure({ adapter: new Adapter() });
describe('Thermometer Component tests', () => {
    it('should render the component', () => {
        const wrapper = shallow(
            <Thermometer
                theme={'light'}
                value={80}
                max={100}
                format={'$'}
                steps={3}
                size={'small'}
                height={180}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});