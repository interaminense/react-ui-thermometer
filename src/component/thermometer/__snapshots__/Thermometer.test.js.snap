// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thermometer Component tests should render the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Thermometer
    format="$"
    height={180}
    max={100}
    size="small"
    steps={3}
    theme="light"
    value={80}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="thermometer__draw-a"
        />,
        <div
          className="thermometer__draw-b"
        />,
        <div
          className="thermometer__meter"
        >
          <ul
            className="thermometer__statistics"
          >
            <li
              style={
                Object {
                  "bottom": "calc(0% - 1px)",
                }
              }
            >
              $0.00
            </li>
            <li
              style={
                Object {
                  "bottom": "calc(33.33% - 1px)",
                }
              }
            >
              $33.33
            </li>
            <li
              style={
                Object {
                  "bottom": "calc(66.67% - 1px)",
                }
              }
            >
              $66.67
            </li>
            <li
              style={
                Object {
                  "bottom": "calc(100% - 1px)",
                }
              }
            >
              $100.00
            </li>
          </ul>
          <div
            className="thermometer__mercury"
            style={
              Object {
                "height": "80%",
              }
            }
          >
            <div
              className="thermometer__percent-current"
            >
              $80
            </div>
            <div
              className="thermometer__mask"
            >
              <div
                className="thermometer__bg-color"
                style={
                  Object {
                    "height": "calc(180px - 57px)",
                  }
                }
              />
            </div>
          </div>
        </div>,
      ],
      "className": "thermometer thermometer--small thermometer--theme-light",
      "style": Object {
        "height": "180px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "thermometer__draw-a",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "thermometer__draw-b",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ul
              className="thermometer__statistics"
            >
              <li
                style={
                  Object {
                    "bottom": "calc(0% - 1px)",
                  }
                }
              >
                $0.00
              </li>
              <li
                style={
                  Object {
                    "bottom": "calc(33.33% - 1px)",
                  }
                }
              >
                $33.33
              </li>
              <li
                style={
                  Object {
                    "bottom": "calc(66.67% - 1px)",
                  }
                }
              >
                $66.67
              </li>
              <li
                style={
                  Object {
                    "bottom": "calc(100% - 1px)",
                  }
                }
              >
                $100.00
              </li>
            </ul>,
            <div
              className="thermometer__mercury"
              style={
                Object {
                  "height": "80%",
                }
              }
            >
              <div
                className="thermometer__percent-current"
              >
                $80
              </div>
              <div
                className="thermometer__mask"
              >
                <div
                  className="thermometer__bg-color"
                  style={
                    Object {
                      "height": "calc(180px - 57px)",
                    }
                  }
                />
              </div>
            </div>,
          ],
          "className": "thermometer__meter",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  style={
                    Object {
                      "bottom": "calc(0% - 1px)",
                    }
                  }
                >
                  $0.00
                </li>,
                <li
                  style={
                    Object {
                      "bottom": "calc(33.33% - 1px)",
                    }
                  }
                >
                  $33.33
                </li>,
                <li
                  style={
                    Object {
                      "bottom": "calc(66.67% - 1px)",
                    }
                  }
                >
                  $66.67
                </li>,
                <li
                  style={
                    Object {
                      "bottom": "calc(100% - 1px)",
                    }
                  }
                >
                  $100.00
                </li>,
              ],
              "className": "thermometer__statistics",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": "$0.00",
                  "style": Object {
                    "bottom": "calc(0% - 1px)",
                  },
                },
                "ref": null,
                "rendered": "$0.00",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": "$33.33",
                  "style": Object {
                    "bottom": "calc(33.33% - 1px)",
                  },
                },
                "ref": null,
                "rendered": "$33.33",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": "$66.67",
                  "style": Object {
                    "bottom": "calc(66.67% - 1px)",
                  },
                },
                "ref": null,
                "rendered": "$66.67",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "3",
                "nodeType": "host",
                "props": Object {
                  "children": "$100.00",
                  "style": Object {
                    "bottom": "calc(100% - 1px)",
                  },
                },
                "ref": null,
                "rendered": "$100.00",
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="thermometer__percent-current"
                >
                  $80
                </div>,
                <div
                  className="thermometer__mask"
                >
                  <div
                    className="thermometer__bg-color"
                    style={
                      Object {
                        "height": "calc(180px - 57px)",
                      }
                    }
                  />
                </div>,
              ],
              "className": "thermometer__mercury",
              "style": Object {
                "height": "80%",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "$80",
                  "className": "thermometer__percent-current",
                },
                "ref": null,
                "rendered": "$80",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="thermometer__bg-color"
                    style={
                      Object {
                        "height": "calc(180px - 57px)",
                      }
                    }
                  />,
                  "className": "thermometer__mask",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "className": "thermometer__bg-color",
                    "style": Object {
                      "height": "calc(180px - 57px)",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="thermometer__draw-a"
          />,
          <div
            className="thermometer__draw-b"
          />,
          <div
            className="thermometer__meter"
          >
            <ul
              className="thermometer__statistics"
            >
              <li
                style={
                  Object {
                    "bottom": "calc(0% - 1px)",
                  }
                }
              >
                $0.00
              </li>
              <li
                style={
                  Object {
                    "bottom": "calc(33.33% - 1px)",
                  }
                }
              >
                $33.33
              </li>
              <li
                style={
                  Object {
                    "bottom": "calc(66.67% - 1px)",
                  }
                }
              >
                $66.67
              </li>
              <li
                style={
                  Object {
                    "bottom": "calc(100% - 1px)",
                  }
                }
              >
                $100.00
              </li>
            </ul>
            <div
              className="thermometer__mercury"
              style={
                Object {
                  "height": "80%",
                }
              }
            >
              <div
                className="thermometer__percent-current"
              >
                $80
              </div>
              <div
                className="thermometer__mask"
              >
                <div
                  className="thermometer__bg-color"
                  style={
                    Object {
                      "height": "calc(180px - 57px)",
                    }
                  }
                />
              </div>
            </div>
          </div>,
        ],
        "className": "thermometer thermometer--small thermometer--theme-light",
        "style": Object {
          "height": "180px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "thermometer__draw-a",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "thermometer__draw-b",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ul
                className="thermometer__statistics"
              >
                <li
                  style={
                    Object {
                      "bottom": "calc(0% - 1px)",
                    }
                  }
                >
                  $0.00
                </li>
                <li
                  style={
                    Object {
                      "bottom": "calc(33.33% - 1px)",
                    }
                  }
                >
                  $33.33
                </li>
                <li
                  style={
                    Object {
                      "bottom": "calc(66.67% - 1px)",
                    }
                  }
                >
                  $66.67
                </li>
                <li
                  style={
                    Object {
                      "bottom": "calc(100% - 1px)",
                    }
                  }
                >
                  $100.00
                </li>
              </ul>,
              <div
                className="thermometer__mercury"
                style={
                  Object {
                    "height": "80%",
                  }
                }
              >
                <div
                  className="thermometer__percent-current"
                >
                  $80
                </div>
                <div
                  className="thermometer__mask"
                >
                  <div
                    className="thermometer__bg-color"
                    style={
                      Object {
                        "height": "calc(180px - 57px)",
                      }
                    }
                  />
                </div>
              </div>,
            ],
            "className": "thermometer__meter",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    style={
                      Object {
                        "bottom": "calc(0% - 1px)",
                      }
                    }
                  >
                    $0.00
                  </li>,
                  <li
                    style={
                      Object {
                        "bottom": "calc(33.33% - 1px)",
                      }
                    }
                  >
                    $33.33
                  </li>,
                  <li
                    style={
                      Object {
                        "bottom": "calc(66.67% - 1px)",
                      }
                    }
                  >
                    $66.67
                  </li>,
                  <li
                    style={
                      Object {
                        "bottom": "calc(100% - 1px)",
                      }
                    }
                  >
                    $100.00
                  </li>,
                ],
                "className": "thermometer__statistics",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "$0.00",
                    "style": Object {
                      "bottom": "calc(0% - 1px)",
                    },
                  },
                  "ref": null,
                  "rendered": "$0.00",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "$33.33",
                    "style": Object {
                      "bottom": "calc(33.33% - 1px)",
                    },
                  },
                  "ref": null,
                  "rendered": "$33.33",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "$66.67",
                    "style": Object {
                      "bottom": "calc(66.67% - 1px)",
                    },
                  },
                  "ref": null,
                  "rendered": "$66.67",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "3",
                  "nodeType": "host",
                  "props": Object {
                    "children": "$100.00",
                    "style": Object {
                      "bottom": "calc(100% - 1px)",
                    },
                  },
                  "ref": null,
                  "rendered": "$100.00",
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="thermometer__percent-current"
                  >
                    $80
                  </div>,
                  <div
                    className="thermometer__mask"
                  >
                    <div
                      className="thermometer__bg-color"
                      style={
                        Object {
                          "height": "calc(180px - 57px)",
                        }
                      }
                    />
                  </div>,
                ],
                "className": "thermometer__mercury",
                "style": Object {
                  "height": "80%",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "$80",
                    "className": "thermometer__percent-current",
                  },
                  "ref": null,
                  "rendered": "$80",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="thermometer__bg-color"
                      style={
                        Object {
                          "height": "calc(180px - 57px)",
                        }
                      }
                    />,
                    "className": "thermometer__mask",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "className": "thermometer__bg-color",
                      "style": Object {
                        "height": "calc(180px - 57px)",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
